module/zfs/zfs_fm.c

```diff
@@ -1444,6 +1444,58 @@ zfs_ereport_fini(void)
 	mutex_destroy(&recent_events_lock);
 }
 
+void
+zfs_ereport_snapshot_post(const char *subclass, spa_t *spa, const char *name)
+{
```

Comment:
Hopefully just this last comment, we've added these new events which we think will be useful on all platforms yet they aren't called anywhere.  As part of this PR I think we want to call them from the common code to exercise this change and so they'll just be there and working when MacOS needs them.

Reply:
symlinks

We use this in place of `mount` - we have zero permission to cause a mount in kernel.

```
                                dprintf("%s: requesting mount for '%s'\n",
                                    __func__, full_name);

                                mutex_enter(&zfsctl_mounts_lock);
                                list_insert_tail(&zfsctl_mounts_list, zcm);
                                mutex_exit(&zfsctl_mounts_lock);

                                mutex_enter(&zcm->zcm_lock);
                                zfs_ereport_snapshot_post(
                                    FM_RESOURCE_ZFS_SNAPSHOT_MOUNT,
                                    dmu_objset_spa(zfsvfs->z_os), full_name);

                                /* Now we wait hoping zed comes back to us */
                                ret = cv_timedwait(&zcm->zcm_cv, &zcm->zcm_lock,
                                    ddi_get_lbolt() + SEC_TO_TICK(6));

```

So do you then want something like:

```
zfsctl_snapshot_mount(struct path *path, int flags)
{
...
        error = call_usermodehelper(argv[0], argv, envp, UMH_WAIT_PROC);
...
        zfs_ereport_snapshot_post(
                                    FM_RESOURCE_ZFS_SNAPSHOT_MOUNT,
                                    dmu_objset_spa(zfsvfs->z_os), full_name);
```

The zedlets would need something like `#if linux // do nothing #else if macOS // "zfs mount" `